src/test/frontend/deploy/createsecret_controller_test.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CreateSecretController from 'deploy/createsecret_controller';
+import deployModule from 'deploy/deploy_module';
+
+describe('Create-Secret dialog', () => {
+  /** @type {!CreateSecretControllerController} */
+  let ctrl;
+  /** @type {!angular.$httpBackend} */
+  let httpBackend;
+
+  beforeEach(() => {
+    angular.mock.module(deployModule.name);
+    angular.mock.inject(($controller, $httpBackend, $mdDialog, $log, _errorDialog_) => {
+      ctrl = $controller(CreateSecretController, {
+        'namespace': 'default',
+        'mdDialog_': $mdDialog,
+        'log_': $log,
+        'errorDialog_': _errorDialog_,
```

Comment:
You dont need to pass the objects that you dont create manuall. Pass only namespace. Everything else will be injected.